client/trino-client/src/main/java/io/trino/client/ClientCapabilities.java

```diff
@@ -24,4 +29,9 @@
     //   interval X(p1) to Y(p2)
     // When this capability is not set, the server returns datetime types with precision = 3
     PARAMETRIC_DATETIME;
+
+    public static Set<String> allCapabilities()
+    {
+        return Arrays.stream(values()).map(Enum::name).collect(toImmutableSet());
```

Comment:
Isn't this redundant over `ClientCapabilities.values()` that everyone can call directly?

Yes, this additionally maps enum<->name, but from-string and to-string conversion should be handled in one layer (so it's not really belonging here)